diff --git a/source/runtime.cpp b/source/runtime.cpp
--- a/source/runtime.cpp
+++ b/source/runtime.cpp
@@ -438,9 +438,15 @@ bool reshade::runtime::on_init()
 
 	const input::window_handle window = get_hwnd();
 	if (window != nullptr && !_is_vr)
+	{
 		_input = input::register_window(window);
+		_primary_input_handler = _input.use_count() == 1;
+	}
 	else
+	{
 		_input.reset();
+		_primary_input_handler = _input_gamepad != nullptr;
+	}
 
 	// GTK 3 enables transparency for windows, which messes with effects that do not return an alpha value, so disable that again
 	if (window != nullptr)
@@ -830,9 +836,9 @@ void reshade::runtime::on_present()
 	apply_state(cmd_list, _app_state);
 
 	// Update input status
-	if (_input != nullptr)
+	if (_primary_input_handler && _input != nullptr)
 		_input->next_frame();
-	if (_input_gamepad != nullptr)
+	if (_primary_input_handler && _input_gamepad != nullptr)
 		_input_gamepad->next_frame();
 
 	// Save modified INI files
diff --git a/source/runtime.hpp b/source/runtime.hpp
--- a/source/runtime.hpp
+++ b/source/runtime.hpp
@@ -261,6 +261,7 @@ namespace reshade
 
 		bool _ignore_shortcuts = false;
 		bool _force_shortcut_modifiers = true;
+		bool _primary_input_handler = false;
 		std::shared_ptr<class input> _input;
 		std::shared_ptr<class input_gamepad> _input_gamepad;
 
diff --git a/source/runtime_gui.cpp b/source/runtime_gui.cpp
--- a/source/runtime_gui.cpp
+++ b/source/runtime_gui.cpp
@@ -916,7 +916,7 @@ void reshade::runtime::draw_gui()
 #endif
 		)
 	{
-		if (_input != nullptr)
+		if (_primary_input_handler && _input != nullptr)
 		{
 			_input->block_mouse_input(_block_input_next_frame);
 			_input->block_keyboard_input(_block_input_next_frame);
@@ -1561,7 +1561,7 @@ void reshade::runtime::draw_gui()
 	// Render ImGui widgets and windows
 	ImGui::Render();
 
-	if (_input != nullptr)
+	if (_primary_input_handler && _input != nullptr)
 	{
 		const bool block_input = _input_processing_mode != 0 && (_show_overlay || _block_input_next_frame);
 		const bool block_mouse_input = block_input && (imgui_io.WantCaptureMouse || _input_processing_mode == 2);
